clang/test/3C/implicit_casts.c

```diff
@@ -0,0 +1,26 @@
+// RUN: 3c -base-dir=%S -alltypes -warn-root-cause %s -- -Xclang -verify -Wno-everything
```

Comment:
Maybe a filename like `implicit_casts_root_cause.c` would make the scope of this file clearer?  We currently have some tests of other 3C behaviors related to implicit casts in other files.